pages/discover/collections.tsx

```diff
@@ -0,0 +1,317 @@
+      result = SORT_OPTION_ORDER[UrlParamKey.TIME_PERIOD].map(
+        (o) => SORT_OPTIONS.subOptions![SortOption.BY_MARKET_CAP].subOptions![o].label
+      );
+    } else {
+      result = [];
+    }
+    return result;
+  }, [urlParams]);
+
+  const menus: JSX.Element[] = [
+    <CardGridWithHeader.HeaderElement key="primary-sort">
+      <DropdownSelect
+        keys={SORT_OPTION_ORDER[UrlParamKey.BY]}
+        onSelect={(k) => urlParamSetters[UrlParamKey.BY](k)}
+        defaultKey={SORT_OPTIONS.defaultSubOptionValue}
+        selectedKey={urlParams[UrlParamKey.BY]}
+      >
+        {primarySortLabels}
+      </DropdownSelect>
+    </CardGridWithHeader.HeaderElement>,
+  ];
+
+  if (urlParams[UrlParamKey.BY] === SortOption.BY_VOLUME) {
+    menus.push(
+      <CardGridWithHeader.HeaderElement key="volume-sort">
+        <DropdownSelect
+          keys={SORT_OPTION_ORDER[UrlParamKey.TIME_PERIOD]}
+          onSelect={(k) => urlParamSetters[UrlParamKey.TIME_PERIOD](k)}
+          defaultKey={SORT_OPTIONS.subOptions![SortOption.BY_VOLUME].defaultSubOptionValue}
+          selectedKey={urlParams[UrlParamKey.TIME_PERIOD]}
+        >
+          {secondarySortLabels}
+        </DropdownSelect>
+      </CardGridWithHeader.HeaderElement>
+    );
+  } else if (urlParams[UrlParamKey.BY] === SortOption.BY_MARKET_CAP) {
+    menus.push(
+      <CardGridWithHeader.HeaderElement key="volume-sort">
+        <DropdownSelect
+          keys={SORT_OPTION_ORDER[UrlParamKey.TIME_PERIOD]}
+          onSelect={(k) => urlParamSetters[UrlParamKey.TIME_PERIOD](k)}
+          defaultKey={SORT_OPTIONS.subOptions![SortOption.BY_MARKET_CAP].defaultSubOptionValue}
+          selectedKey={urlParams[UrlParamKey.TIME_PERIOD]}
+        >
+          {secondarySortLabels}
+        </DropdownSelect>
+      </CardGridWithHeader.HeaderElement>
+    );
+  }
+
+  return (
+    <CardGridWithHeader<CollectionPreviewCardData>
+      cardContext={{
+        noDataFallback: <div>No matching Collections</div>,
+        cardCreator: (data, refetch, loading) => (
+          <CollectionPreviewCard context={{data: data, refetch: refetch, loading: loading, fetchMore: () => {}}} />
+        ),
+        loadingCardCreator: () => <CollectionPreviewLoadingCard />,
+      }}
+      dataContext={{
+        data: queryContext.data,
+        refetch: queryContext.refetch,
+        onLoadMore: onLoadMore,
+        hasMore: queryContext.hasMore,
+        loading: queryContext.loading,
+      }}
+      search={{
+        onChange: (v) => urlParamSetters[UrlParamKey.SEARCH](v.trim()),
+        debounceTimeout: SEARCH_DEBOUNCE_TIMEOUT_MS,
+        placeholder: "Search collections"
+      }}
+      menus={menus}
+    />
+  );
+}
+
+DiscoverCollectionsTab.getLayout = function getLayout(
+  discoverPage: DiscoverPageProps & { children: JSX.Element }
+): JSX.Element {
+  return <DiscoverLayout content={discoverPage.children} />;
+};
+
+function useQuery(
```

Comment:
I've intentionally structured the code to remove the apollo/graph-ql types and hooks from these more interior parts of the app. All the Apollo stuff now lives in `discover.hooks.ts` and (IMO) should stay out of components and other interior code. Since this is a local function that should never be exported (see `useApolloQuery` for that) then I think the generic name is more friendly.